### ISSUE TITLE
feat: add Docker .devcontainer setup in docs

### ISSUE THREAD

**Issue by Nsttt** [t3-oss/create-t3-app#1643]
### Is your feature request related to a problem? Please describe.

I often find myself needing to ensure that my development environment is consistent across my team to prevent "it works on my machine" issues. However, setting up a consistent environment manually can be time-consuming and error-prone.

### Describe the solution you'd like to see

I propose adding a Docker .devcontainer setup in the documentation of create-t3-app. This setup would provide a containerized development environment that anyone can use regardless of their local machine setup. By following the steps in the documentation, developers would be able to quickly get a consistent, reproducible development environment up and running.

### Describe alternate solutions

Don't do it.

**Comment by juliusmarminge** [comment]
does hmr and all that works in dev with this?

i'd be ok with adding this to the documentation - should it be it's separate document or added as a "sidenote" in the existing docker documentation?

**Comment by Nsttt** [comment]
Yeah, it seems to work just fine. I'm going to give it a bit more thought on how to properly set it up on https://github.com/Nsttt/t3-devcontainer

Any feedback is appreciated.
